open_instruct/dataset_transformation.py

```diff
@@ -1787,9 +1787,13 @@ def get_cached_dataset_tulu_with_statistics(
         )
     elif dataset_cache_mode == "hf":
         cache = DatasetTransformationCache(config_hash=dataset_config_hash, hf_entity=hf_entity)
-    return cache.load_or_transform_dataset(
-        dcs, tc, dataset_skip_cache=dataset_skip_cache, return_statistics=return_statistics
-    )
+
+    dataset, statistics = cache.load_or_transform_dataset(dcs, tc, dataset_skip_cache=dataset_skip_cache)
+
+    if return_statistics:
```

Comment:
I don't really like mutable returns with python functions, because it makes debugging what a function is doing down the line much harder. Can we instead keep the `get_cached_dataset_tulu` as before, and then the statistics script just uses the sub-function that also returns the stats?

That or we set it so wherever the dataset function is called, we also take the statistics out too (which may be none).

Reply:
That's fine with me! updated